File: Code/Python/780. Reaching Points.py
class Solution:
    def reachingPoints(self, sx: int, sy: int, tx: int, ty: int) -> bool:
        while tx >= sx and ty >= sy:
            if tx == sx:
                return (ty - sy) % tx == 0
            if ty == sy:
                return (tx - sx) % ty == 0
            if tx > ty:
                tx %= ty
            else:
                ty %= tx
        return False

File: Code/C++/780. Reaching Points.cpp
class Solution {
public:
    bool reachingPoints(int sx, int sy, int tx, int ty) {
        // from target go backwards to start
        while (tx >= sx && ty >= sy){
            if ((tx == sx && (ty - sy) % tx == 0) || (ty == sy && (tx - sx) % ty == 0)){
                return true;
            }
            if (tx > ty){
                tx %= ty;
            }
            else {
                ty %= tx;
            }

        }
        return false;

    }
};